openbas-front/src/components/common/chaining/ChainingUtils.tsx

```diff
@@ -15,6 +22,24 @@ const fromInjectDependencyToInputDependency = (dependencies: InjectDependency[])
   return dependencies.length > 0 ? result : null;
 };
 
+const fromInjectDependencyToLabel = (dependency: string) => {
+  let label = '';
+  const splittedConditions = dependency.split(breakpointAndOr);
+  for (let i = 0; i < splittedConditions.length; i += 1) {
+    if (splittedConditions[i].trim() === '&&') {
+      label += ' AND ';
+    } else if (splittedConditions[i].trim() === '||') {
+      label += ' OR ';
+    } else {
+      const splittedValues = splittedConditions[i].trim().split(breakpointValue);
+      const key = Array.from(splittedValues[0].trim().matchAll(typeFromName), (m) => m[1]);
+      label += `${key} is ${splittedValues[1].trim() === 'true' ? 'Success' : 'Failure'}`;
+    }
+  }
+
+  return label;
+};
```

Comment:
**suggestion (non-blocking):** when possible, it's preferable to use a functional approach, since immutability and no side-effects reduce bugs and it can be more easy to understand code

here we could use `reduce` (not tested):
```
const fromInjectDependencyToLabel = (dependency: string) => {
  const splittedConditions = dependency.split(breakpointAndOr);

  return splittedConditions.reduce((label, condition) => {
    const trimmedCondition = condition.trim();

    if (trimmedCondition === '&&') {
      return label + ' AND ';
    } else if (trimmedCondition === '||') {
      return label + ' OR ';
    } else {
      const splittedValues = trimmedCondition.split(breakpointValue);
      const key = Array.from(splittedValues[0].trim().matchAll(typeFromName), (m) => m[1]);
      return label + `${key} is ${splittedValues[1].trim() === 'true' ? 'Success' : 'Failure'}`;
    }
  }, '');
};
```

Reply:
Good point ! Doing :)